src/platform/modulesmanager/src/ModulesManager.cpp

```diff
@@ -88,10 +100,16 @@ MPI_HANDLE MpiOpen(
 
 void MpiClose(MPI_HANDLE clientSession)
 {
-    if (mpiMap.end() != mpiMap.find(clientSession))
+    ModulesManager* modulesManager = reinterpret_cast<ModulesManager*>(clientSession);
+
+    if ((nullptr != modulesManager) && (mpiMap.end() != mpiMap.find(modulesManager->GetClientName())))

```

Comment:
Let's imagine this sequence, all for same client:

1. MpiOpen
2. MpiGetReported
3. MpiClose

The 2nd call increments the ref count on the session opened by 1st then decrements it. The 3rd call decrements the ref count once more and when the ref count goes to 0 the MM object for this client session self-destroys itself.

Consider using ref counting and self-release on these objects, COM style?